core/agent_parser.go

```diff
@@ -134,35 +150,107 @@ func bzeroInit(log logger.T) {
 	}
 	data, err := json.Marshal(keys)
 	if err != nil {
-		log.Errorf("BZero Marshalling of Keys Failed: %v", err)
-		os.Exit(1)
+		log.Errorf("Marshalling BZero Keys Failed: %v", err)
+		return bzeroreg.BZeroRegErrorExitCode
 	}
 
-	if err = vault.Store(keysplitting.BZeroConfig, data); err != nil {
-		log.Errorf("BZero Storing of Config Failed: %v", err)
-		os.Exit(1)
+	if err = vault.Store(bzeroreg.BZeroConfigStorage, data); err != nil {
+		log.Errorf("Storing Bzero Config Failed: %v", err)
+		return bzeroreg.BZeroRegErrorExitCode
+	}
+
+	log.Info("Successfully initialized and stored all BastionZero variables")
+	return 0
+}
+
+func bzeroRegistration(log logger.T) (exitCode int) {
+	// Make registration endpoint API calls
+	log.Info("Making registration request to BastionZero...")
+	resp, err := bzeroreg.Register(log, bzeroRegistrationKey, bzeroEnvName, bzeroEnvID, bzeroTargetName, bzeroServiceUrl)
+	if err != nil {
+		log.Infof("error registering: %s", err)
+		return bzeroreg.BZeroRegErrorExitCode
 	}
+	log.Info("Registration call to BastionZero successful!")
 
-	log.Info("Successfully created and stored BZero Config!")
+	// save registration response variables locally
+	activationCode = resp.ActivationCode
+	activationID = resp.ActivationId
+	region = resp.ActivationRegion
+
+	// Only assign either or both of the SSO org fields, if they were not provided
```

Comment:
not sure if I understand this comment , I think it should read:
`Only assign SSO org ID and provider fields if they were not provided`

Reply:
7193eee